README.rst

```diff
@@ -1,334 +1,503 @@
 
 
 Installation
-------------
+============
+
+.. code:: bash
+
+   pip install pytest-datafiles
+
+
+Upgrade to 3.0
+==============
+
+Version 3 now uses `tmp_path
+<https://docs.pytest.org/en/latest/tmp_path.html>`_, resulting in
+*pathlib.Path* objects instead of *py.path*.
+
+Your tests may need to be adjusted.
+
+.. code:: python
+
+   """Upgrade to v3."""
+   import os
+
+   import pytest
+   import py
+
 
-.. code-block:: bash
+   @pytest.mark.datafiles('/tmp/big_files/film1.mp4')
+   def test_convert_to_pypath(datafiles):
+       """Convert `datafiles` to `py.path` in place.
 
-    pip install pytest-datafiles
+       Changing the type back to py.path - should be avoided!
+
+       Better to update the tests to make use of `pathlib`.
+       """
+
+       # changing the type back to py.path - should be avoided
+       datafiles = py.path.local(str(datafiles))
+
+       # code remains as before
+       assert len(datafiles.listdir()) == 1
+       assert (datafiles / 'film1.mp4').check(file=1)
+
+
+   @pytest.mark.datafiles('/tmp/big_files/film1.mp4')
+   def test_update_to_pathlib(datafiles):
+       """Upgrade code to use `pathlib`.
+
+       Some examples showing `pathlib` style.
+       """
+
+       # For demo purpose: this is the old-style
+       pypath_datafiles = py.path.local(str(datafiles))
+
+       # old style
+       assert len(pypath_datafiles.listdir()) == 1
+       assert (pypath_datafiles / 'film1.mp4').check(file=1)
+
+       # new style  # pylint: disable=R0801
+       assert len(list(datafiles.iterdir())) == 1
+       assert (datafiles / 'film1.mp4').is_file()
+
+
+   @pytest.mark.datafiles('/tmp/big_files/film1.mp4')
+   def test_keep_using_str(datafiles):
+       """Keep using plain `os`.
+
+       Nothing changes.
+       """
+
+       path = str(datafiles)
+       assert len(os.listdir(path)) == 1
+       assert os.path.isfile(os.path.join(path, 'film1.mp4'))
 
 
 Usage
------
+=====
+
+These examples can also be found in *tests/examples*.
+
 
 Example 1
-~~~~~~~~~
+---------
+
+One possible use case is when you are running tests on very big files
+that are not included or packaged with your tests. For example, your
+test files are large video files stored under */opt/big_files/* . You
+don’t want your tests modifying the original files, but the files are
+required by the tests. You can reference these data files in your test
+method as follows:
+
+.. code:: python
+
+   """Example: Reference files anywhere """
+   import os
+   import pytest
 
-One possible use case is when you are running tests on very big files that are
-not included or packaged with your tests. For example, your test files are
-large video files stored under */opt/big_files/* . You don't want your tests modifying
-the original files, but the files are required by the tests. You can reference these
-data files in your test method as follows:
 
-.. code-block:: python
+   @pytest.mark.datafiles('/tmp/big_files/film1.mp4')
```

Comment:
I assume you are using /tmp/ here because you made the examples actually executable (which is a good idea to ensure they keep working) but it has the disadvantage that using /tmp/ makes the purpose of this plugin much more confusing to users. If you already have the files in /tmp/ why use this plugin? How do the files get into /tmp? What if they are no longer there (because /tmp is not a trustworthy source)? etc.

I don't have a good solution right now. It's a trade-off. Just wanted to note it in case you can come up with a better idea.

Reply:
See the next comment for a solution.

Reply:
done